core/mondoo-ms365-security.mql.yaml

```diff
@@ -0,0 +1,625 @@
+        properties['passwordMinimumLength'] == 6 } msgraph.beta.devicemanagement.deviceConfigurations.where( properties["@odata.type"] == "#microsoft.graph.macOSGeneralDeviceConfiguration") {
+        properties['passwordMinimumLength'] == 6 } msgraph.beta.devicemanagement.deviceConfigurations.where( properties["@odata.type"] == "#microsoft.graph.iosGeneralDeviceConfiguration") {
+        properties['passcodeMinimumLength'] == 6 } msgraph.beta.devicemanagement.deviceConfigurations.where( properties["@odata.type"] == "#microsoft.graph.androidGeneralDeviceConfiguration") {
+        properties['passwordMinimumLength'] == 6 } msgraph.beta.devicemanagement.deviceConfigurations.where( properties["@odata.type"] == "#microsoft.graph.androidWorkProfileGeneralDeviceConfiguration") {
+        properties['passwordMinimumLength'] == 6 }") }" --certificate-path <*.pem> --tenant-id <tenant_id> --client-id <client_id>
+          ```
+        __cnquery shell__
+
+        To audit Microsoft ms365 with `cnquery shell`:
+
+        1. Launch `cnquery shell`:
+
+          ```bash
+          cnquery shell ms365 --certificate-path <*.pem> --tenant-id <tenant_id> --client-id <client_id>
+          ```
+        
+        2. Run the following query:
+
+          ```mql
+                msgraph.beta.devicemanagement.deviceConfigurations.where( properties["@odata.type"] == "#microsoft.graph.windows10GeneralConfiguration") {
+        properties['passwordMinimumLength'] == 6 } msgraph.beta.devicemanagement.deviceConfigurations.where( properties["@odata.type"] == "#microsoft.graph.macOSGeneralDeviceConfiguration") {
+        properties['passwordMinimumLength'] == 6 } msgraph.beta.devicemanagement.deviceConfigurations.where( properties["@odata.type"] == "#microsoft.graph.iosGeneralDeviceConfiguration") {
+        properties['passcodeMinimumLength'] == 6 } msgraph.beta.devicemanagement.deviceConfigurations.where( properties["@odata.type"] == "#microsoft.graph.androidGeneralDeviceConfiguration") {
+        properties['passwordMinimumLength'] == 6 } msgraph.beta.devicemanagement.deviceConfigurations.where( properties["@odata.type"] == "#microsoft.graph.androidWorkProfileGeneralDeviceConfiguration") {
+        properties['passwordMinimumLength'] == 6 } 
+          ```
+      remediation: |
+        ### MS365 Console
+         
+         To update via the ms365 portal:
+
+         1. Log in as to the ms365 portal at https://admin.microsoft.com
+         2. Endpoint Manager --> Devices --> Policy --> Configuration profiles 
+         3. Ensure that a profile exists for each Platform with following conditions:
+            * Password section --> Device restrictions --> Minimum password length is set to 6
+    query: |
+      msgraph.beta.devicemanagement.deviceConfigurations.where( properties["@odata.type"] == "#microsoft.graph.windows10GeneralConfiguration") {
```

Comment:
`msgraph.beta` is deprecated, use latest mql resources

Reply:
Thanks Chris, done!